### PR TITLE
Dataflow runner options: disk type & streaming engine

### PR DESCRIPTION
**Pull request by pyalex**
**What this PR does / why we need it**:

This PR extends execution parameteres for Dataflow Job according to [official documentation](https://cloud.google.com/dataflow/docs/guides/specifying-exec-params#setting-other-cloud-dataflow-pipeline-options).

Next parameter were added:
* workerDiskType
* enableStreamingEngine

Next parameter were updated:
* zone -> workerZone (due to deprecation)

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

new options introduced in application.yaml

feast:
  jobs:
    runners:
    - name: dataflow
      options:
        enableStreamingEngine: false
        workerDiskType: compute.googleapis.com/projects/asia-east1-a/diskTypes/pd-ssd
        workerZone: asia-east1-a  # renamed "zone" due to deprecation

```

### CODE REVIEW

File: core/src/main/resources/application.yml [hunk 1]
```diff
@@ -42,11 +42,13 @@ feast:
         options:
           project: my_gcp_project
           region: asia-east1
-          zone: asia-east1-a
+          workerZone: asia-east1-a
```
Comment (woop) [inline core/src/main/resources/application.yml:45]: Should this be in the release notes?

Comment (pyalex) [reply core/src/main/resources/application.yml:45]: added

**Pull request merged**